module/zfs/zil.c

```diff
@@ -2950,7 +2922,7 @@ zil_commit_waiter_timeout(zilog_t *zilog, zil_commit_waiter_t *zcw)
 	 */
 	lwb_t *nlwb = zil_lwb_write_issue(zilog, lwb);
 
-	IMPLY(nlwb != NULL, lwb->lwb_state != LWB_STATE_OPENED);
+	ASSERT3S(lwb->lwb_state, !=, LWB_STATE_OPENED);
```

Comment:
Or even the following according to the existing code.

```
ASSERT3S(lwb->lwb_state, ==, LWB_STATE_ISSUED);
```

Reply:
Sure.